src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let result = [];
+  const copyState = Object.assign({}, state);
+  for (const obj of actions) {
+    switch (obj.type) {
+      case "addProperties":
+        for (const key in obj.extraData) {
+          copyState[key] = obj.extraData[key];
+        }
+        result.push(Object.assign({}, copyState));
+        break;
+
+      case "removeProperties":
+        for (const remove of obj.keysToRemove) {
+          delete copyState[remove];
+        }
+        result.push(Object.assign({}, copyState));
+        break;
+
+      default:
+        for (const del in copyState) {
+          delete copyState[del];
+        }
+        result.push(Object.assign({}, copyState));
```

Comment:
DRY - don't repeat yourself
just make a push after you switch
```suggestion
        break;

      case "removeProperties":
        for (const remove of obj.keysToRemove) {
          delete copyState[remove];
        }
        break;

      default:
        for (const del in copyState) {
          delete copyState[del];
        }
      }
      
      result.push(Object.assign({}, copyState));

```